afs-core/src/test/java/com/powsybl/afs/AfsBaseTest.java

```diff
@@ -255,6 +257,161 @@ public void archiveAndUnarchiveTestWithZip() throws IOException {
         assertEquals(1, dir2.getChildren().size());
     }
 
+    @Test
+    public void archiveAndUnarchiveTestWithDependencies() throws IOException {
+        /* In this test, there are two directories each with a file
+           There is a dependency between the first and the second file
+           Test archive and unarchive the first directory with these dependencies */
+
+        Project project = afs.getRootFolder().createProject("test");
+        ProjectFolder rootFolder = project.getRootFolder();
+        ProjectFolder dir1 = rootFolder.createFolder("dir1");
+        ProjectFolder dir2 = rootFolder.createFolder("dir2");
+
+        NodeInfo testDataInfo = storage.createNode(dir1.getId(), "data", "data", "", 0, new NodeGenericMetadata());
+        NodeInfo testData2Info = storage.createNode(dir2.getId(), "data2", "data", "", 0,
+                new NodeGenericMetadata().setString("s1", "v1")
+                        .setDouble("d1", 1d)
+                        .setInt("i1", 2)
+                        .setBoolean("b1", false));
+        storage.setConsistent(testDataInfo.getId());
+        storage.setConsistent(testData2Info.getId());
+        storage.addDependency(testDataInfo.getId(), "mylink2", testData2Info.getId());
+
+        Path rootDir = Files.createTempDirectory("testDir");
```

Comment:
You should use a in-memory file system. Please do not write files directly on the file system.

Reply:
I'm agree with that but I can't because when I zip, I needs function toFile() and Jimfs don't implement this function.

Reply:
If you need to get a File, you probably have a design issue.